---
title: "Step 2"
author: "Lisa Leung"
date: '2018-03-03'
output: github_document
---
#Step 2 - eQTM

```{r}
library(tidyverse)
library(reshape)
library(ggplot2)
m <- load("~/STAT540/Repo_team_Gene_Heroes/Code/Step-2/rosmap_postprocV2.rdata")
probes_subjects[1:5,1:5]
subjects_genes[1:5,1:5]

rownames(subjects_genes)==names(probes_subjects)
# cor.test(subjects_genes[,1],as.numeric(t(probes_subjects[1,])))$p.value

dat <- as.data.frame(as.matrix(probes_genes_distance))
dim(dat)
colnames(dat) <- names(subjects_genes)
rownames(dat) <- rownames(probes_subjects)
```

Adjust PCs for gene expression data (labelled as m)

```{r}
#SVD PCA
pc_svd <- svd(subjects_genes)

#exploring the dimensionality on PCA
length(pc_svd$d)
dim(pc_svd$u) #u is equivalent to the principal components
dim(pc_svd$v) 
plot(pc_svd$u[, 1], pc_svd$u[, 2], main = "SVD", xlab = "U1", ylab = "U2") #plot the first two PCs

# reconstructing the same values is shown below:
# SVD$u %*% diag(SVD$d) %*% t(SVD$v)

# Adjusting for the PCs according to number of PCA you want to adjust in gene expression data
m <- list()
n <- c(1,5,10,15,20, 25, 30)
for (i in n){
diag_of_d <- pc_svd$d
diag_of_d[1:i] <- 0
m[[i]] <- pc_svd$u %*% diag(diag_of_d) %*% t(pc_svd$v)
rownames(m[[i]]) <- rownames(subjects_genes)
colnames(m[[i]]) <- colnames(subjects_genes)
}

saveRDS(m, "subjects_genes_PCA_adjusted_V3.RDS")
```

Below is adjusting for the DNA methylation probes (labelled as m2)
```{r}
#SVD PCA
pc_svd_probes <- svd(t(probes_subjects))

#exploring the dimensionality on PCA
length(pc_svd_probes$d)
dim(pc_svd_probes$u) #u is equivalent to the principal components
dim(pc_svd_probes$v) 
plot(pc_svd_probes$u[, 1], pc_svd$u[, 2], main = "SVD", xlab = "U1", ylab = "U2") #plot the first two PCs

# reconstructing the same values is shown below:
# SVD$u %*% diag(SVD$d) %*% t(SVD$v)

# Adjusting for the PCs according to number of PCA you want to adjust
m2 <- list()
for (i in n){
diag_of_d <- pc_svd_probes$d
diag_of_d[1:i] <- 0
m2[[i]] <- pc_svd_probes$u %*% diag(diag_of_d) %*% t(pc_svd_probes$v)
rownames(m2[[i]]) <- colnames(probes_subjects)
colnames(m2[[i]]) <- rownames(probes_subjects)
}

saveRDS(m2, "probes_subjects_PCA_adjusted_V3.RDS")
```

Look at random 50 genes on number of significant results on:

1, 5, 10, 15, 20, 25, 30 adjusted PCs in DNA methylation probe
1, 5, 10, 15, 20, 25, 30 adjusted PCs in gene expression data 

This include plots with the number of PCs adjusted in either data set with the proportion of genes with at least one significant hit.

```{r}
set.seed(100)
n_genes <- sample(1:ncol(subjects_genes),50)

results_pca_50_genes = lapply(n, function(k){
out_per_gdata <-  lapply(n, function(l){
out_per_gene <-  lapply(n_genes, function(i){
    toLoop = which(dat[,i] != 0) 
    out <- lapply(toLoop, function(j){
      data.frame(probe = rownames(probes_subjects)[j], 
                 gene = colnames(subjects_genes)[i],
                estimate = cor.test(m[[k]][,i],
                                    m2[[l]][,j])$estimate,
                pvalue = cor.test(m[[k]][,i],
                                    m2[[l]][,j])$p.value)
      })
    gene_out <- do.call(rbind, out)
    gene_out$bonferroni <- p.adjust(gene_out$pvalue, method = "bonferroni")
    data.frame(num_PC_gdata=k, gene = colnames(m[[k]])[i], num_less_pvalue=sum(gene_out$bonferroni <= 0.05))
   # num_less_pvalue=sum(gene_out$pvalue <= 0.1))
}) 
do.call(rbind, out_per_gene) %>% mutate(num_PC_probedata = l)
})
do.call(rbind, out_per_gdata)
})

toPlot_50_genes_PCA_gdata_probedata <- do.call(rbind, results_pca_50_genes)
toPlot_50_genes_PCA_gdata_probedata$eqtl<-0
toPlot_50_genes_PCA_gdata_probedata$eqtl[toPlot_50_genes_PCA_gdata_probedata$num_less_pvalue>0] <- 1
saveRDS(toPlot_50_genes_PCA_gdata_probedata, "toPlot_50_genes_PCA_gdata_probedata_V2.RDS")
toPlot_50_genes_PCA_gdata_probedata_average <- toPlot_50_genes_PCA_gdata_probedata %>% group_by(num_PC_gdata, num_PC_probedata) %>% summarise(average_sig = mean(eqtl))
toPlot_50_genes_PCA_gdata_probedata_average$num_PC_gdata <- as.factor(toPlot_50_genes_PCA_gdata_probedata_average$num_PC_gdata )
toPlot_50_genes_PCA_gdata_probedata_average$num_PC_probedata <- as.factor(toPlot_50_genes_PCA_gdata_probedata_average$num_PC_probedata )

jpeg("prop_of_genes_sig_on_50_genes_V3.jpg")
p <- ggplot(aes(x=num_PC_gdata, y = num_PC_probedata), data = toPlot_50_genes_PCA_gdata_probedata_average) + geom_tile(aes(fill=average_sig), color = "white") + scale_fill_gradient(low= "white", high = "steelblue", name = "Proportion of genes significant") 
p +  ggtitle("Proportion of genes with a significant eQTM for 50 genes") + xlab("Number of PCs Adjusted for Gene Expression Data") + ylab("Number of PCs Adjusted for DNA Methylation Data")
dev.off()

```

Let's look at some random genes and look at what happens individually

```{r}
#finding genes in data set
gene_name <- unique(sub(':.*$',"",colnames(subjects_genes)))
sum(gene_name %in% c("AKT2","APOBEC3B","APP"))
colnames(subjects_genes)[grep("AKT2",colnames(subjects_genes))]
colnames(subjects_genes)[grep("APP",colnames(subjects_genes))]

## using lapply looking at number of bonferroni = 0.05 based on number of PCA - recommended
#AKT2 on gene
results_pca_AKT2 = lapply(1:n, function(i){
    toLoop = which(dat[,"AKT2:ENSG00000105221.10"] != 0) 
    out <- lapply(toLoop, function(j){
      data.frame(probe = rownames(probes_subjects)[j], 
                estimate = cor.test(m[[i]][,"AKT2:ENSG00000105221.10"],
                                    as.numeric(t(probes_subjects[j,])))$estimate,
                pvalue = cor.test(m[[i]][,"AKT2:ENSG00000105221.10"], as.numeric(t(probes_subjects[j,])))$p.value)
      })
   gene_out <-  do.call(rbind,out)
   #gene_out$fdr <- p.adjust(gene_out$pvalue, method="fdr")
   data.frame(num_PC=i, num_less_pvalue=sum(gene_out$pvalue <= 0.1))
})

lapply_results_pca_AKT2 <- do.call(rbind,results_pca_AKT2)
ggplot(aes(x=num_PC, y=num_less_pvalue), data = lapply_results_pca_AKT2) + geom_point() + geom_line( ) + ggtitle("Number of DNA probes less than pvalue = 0.1 on number of PCs for AKT2") + xlab("Number of PCs") + ylab("Number of significant DNA probes for alpha at 0.1")

#APP on gene
results_pca_APP = lapply(1:n, function(i){
    toLoop = which(dat[,"APP:ENSG00000142192.14"] != 0) 
    out <- lapply(toLoop, function(j){
      data.frame(probe = rownames(probes_subjects)[j], 
                estimate = cor.test(m[[i]][,"APP:ENSG00000142192.14"],
                                    as.numeric(t(probes_subjects[j,])))$estimate,
                pvalue = cor.test(m[[i]][,"APP:ENSG00000142192.14"], as.numeric(t(probes_subjects[j,])))$p.value)
      })
   gene_out <-  do.call(rbind,out)
   #gene_out$fdr <- p.adjust(gene_out$pvalue, method="fdr")
   data.frame(num_PC=i, num_less_pvalue=sum(gene_out$pvalue <= 0.1))
})

lapply_results_pca_APP <- do.call(rbind,results_pca_APP)
plot(lapply_results_pca_APP$num_PC, lapply_results_pca_APP$num_less_pvalue)

#CCDC106 on gene
results_pca_CCDC106 = lapply(1:n, function(i){
    toLoop = which(dat[,"CCDC106:ENSG00000173581.3"] != 0) 
    out <- lapply(toLoop, function(j){
      data.frame(probe = rownames(probes_subjects)[j], 
                estimate = cor.test(m[[i]][,"CCDC106:ENSG00000173581.3"],
                                    as.numeric(t(probes_subjects[j,])))$estimate,
                pvalue = cor.test(m[[i]][,"CCDC106:ENSG00000173581.3"], as.numeric(t(probes_subjects[j,])))$p.value)
      })
   gene_out <-  do.call(rbind,out)
   #gene_out$fdr <- p.adjust(gene_out$pvalue, method="fdr")
   data.frame(num_PC=i, num_less_pvalue=sum(gene_out$pvalue <= 0.1))
})

lapply_results_pca_CCDC106 <- do.call(rbind,results_pca_CCDC106)
plot(lapply_results_pca_CCDC106$num_PC, lapply_results_pca_CCDC106$num_less_pvalue)


## using for loop - not recommended: below is for our own references 

#res <- data.frame()
#for (i in 1:ncol(subjects_genes)){
#  m <- data.frame()
#  toLoop = which(dat[,i] != 0) 
  
#  for (j in toLoop){
#    print(paste0(i," and ",j))
#    m <- rbind(m, data.frame(gene = names(subjects_genes)[i], probe = rownames(probes_subjects)[j], estimate = #cor.test(subjects_genes[,i], as.numeric(t(probes_subjects[j,])))$estimate,
#pvalue = cor.test(subjects_genes[,i], as.numeric(t(probes_subjects[j,])))$p.value))
#  }
#  res <- rbind(res, m)
#  rm(m)
#}

```

Below is the Cumulative variation explained by PCs

```{r}
#ref:http://genomicsclass.github.io/book/pages/pca_svd.html
#cum all
jpeg("cum_var_explained_all_V2.jpg")
plot(cumsum(pc_svd$d^2/sum(pc_svd$d^2)), xlim = c(0, 481), type = "b", pch = 16, xlab = "principal components", 
     ylab = "cumulative variance explained")
dev.off()

#cum of first 50
jpeg("cum_var_explained_50_V2.jpg")
cumsum(pc_svd$d^2/sum(pc_svd$d^2))[1:50] #total variance seems to stabilize at 47th pc with 78% of variation explained
plot(cumsum(pc_svd$d^2/sum(pc_svd$d^2)), xlim = c(0, 50), type = "b", pch = 16, xlab = "principal components", 
     ylab = "cumulative variance explained")
dev.off()

```

Let's do correlation tests based on the PC adjusted data sets

```{r}
## using lapply - recommended
PC_probes <- 1 # of PCs adjusted for probes data set
PC_genes <- 30 # of PCs adjusted for genes data set

results_PCA = lapply(1:ncol(m[[3]]), function(i){
    toLoop = which(dat[,i] != 0) 
    out <- lapply(toLoop, function(j){
      data.frame(gene = colnames(m[[3]])[i],
                probe = rownames(probes_subjects)[j], 
                estimate = cor.test(m[[3]][,i],
                                    as.numeric(t(probes_subjects[j,])))$estimate,
                pvalue = cor.test(m[[3]][,i], as.numeric(t(probes_subjects[j,])))$p.value)
      })
    do.call(rbind,out)
})

lapply_results_PCA <- do.call(rbind,results_PCA)
lapply_results_PCA$adjusted.pvalue <- p.adjust(lapply_results_PCA$pvalue, method="fdr") #adjusting for fdr
saveRDS(lapply_results_PCA,"cor_test_results_PCA_lapply_V3.rds")

jpeg("histogram_pvalue.jpeg")
hist(lapply_results_PCA$pvalue, main = "Histogram of p-value on correlation tests")
dev.off()

summary(lapply_results_PCA$adjusted.pvalue)
min(lapply_results_PCA$adjusted.pvalue)
```



