---
title: "Step4-Biological Analysis-part 1"
author: "Zohreh Sharafian"
date: "March 24, 2018"
output: github_document
---
We load related library packages to understand and verify the biological aspects of our analysis.
```{r setup, include=FALSE}

library(reshape2)
library(tidyverse)
library(GEOquery)
library(dplyr)
library(RCurl)
library(knitr)
library(purrr)
library(data.table)
```
### Loading data
We load dataset from Step 1:probes_subjects that contains DNA methylation probe values, subjects_genes that contains gene expression values, and probes_genes_distance that is a sparse matrix containing the distances between a gene and a probe.Note that probes_genes_distance is 0 when probe and gene distance is over 1Mb.

```{r}
RosmapData <- load("C:/Users/zohre/Desktop/Repo_team_Gene_Heroes/Data/rosmap_postprocV2.RData")
probes_subjects[1:5, 1:5]
subjects_genes[1:5,1:5]
str(RosmapData)
```
```{r}
nrow(probes_genes_distance) #number of probes
ncol(probes_genes_distance) #number of genes
```
WE also load a dataset from eQTM analysis that we have peroformed in Step 2, including all the adjusted P-values for analyzing single probes.
```{r}
cor_test_results_PCA_lapply_V4 <- readRDS("C:/Users/zohre/Desktop/Repo_team_Gene_Heroes/Data/cor_test_results_PCA_lapply_V4.rds")
head (cor_test_results_PCA_lapply_V4)

```
### Setting a cut of 0.05 for "adjusted P-value"
Next, We extract three columns including gene, probe, and adjusted P value and filtering the genes based on adjusted P-value that are greater than 0.05.

```{r}

extracted_matrix <- cor_test_results_PCA_lapply_V4 [,c(1,2, 5)]
head(extracted_matrix)
nrow(extracted_matrix)

# cut off (0.05) for the p value
filtered_pvalue <- extracted_matrix[extracted_matrix$adjusted.pvalue<0.05,]

```


### Merging two data sets
We create a new data set with the subject_genes as columns and probe_subjects as rows of data.Then, we match the probes and genes from two data sets.Finally, we plot the histogram of our data to see the distance of the methylated CpG sites from transcription start site (TSS) of the genes.     
```{r}

data <- as.data.frame(as.matrix(probes_genes_distance))
colnames(data) <- names(subjects_genes)
rownames(data) <- rownames(probes_subjects)

```

```{r}

#matching the probes and genes from two data set

match_probes <- match(filtered_pvalue$probe, rownames(data) )
matche_genes <- match(filtered_pvalue$gene, colnames(data) )

inds=cbind(match_probes,matche_genes)

match_data <- data[inds]



qplot(match_data,
      geom="histogram",
      main = "The distance of methylated CpG sites from Trascription Start Sites", 
      xlab = "methylated CpG sites distance from TSS",  
      fill=I("blue"), 
      col=I("red"), 
      ) +theme_minimal() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))



```

**The histogram shows that the majority of methylated CpG sites are concentrated near the transcription start sites (TSS) of genes.** 


Next, we get the mean of adjusted P-value for each bin of histogram (number of methylated CpG sites) to check the distance of the significant probes from TSS.  

```{r} 

tmp <- hist(match_data) 

intervals_values=findInterval(match_data,tmp$breaks)  #get the interval values from the histogram


```

Using a for loop function to get the interval values for all the genes. Then, we generate a genome-line plot of adjusted P-value for all the methylated CpG sites in the genes promoters.

```{r}
histog <-  integer (10)
for(i in 1:10) {
elemenets_in_this_interval <-which(intervals_values==i);
histog[i]=mean(filtered_pvalue$adjusted.pvalue[elemenets_in_this_interval])
print((histog[i]))
}
x=tmp$mids;
y=histog;
data_plot=cbind(x,y)
data_plot=as.data.frame(data_plot)
colnames(data_plot)= c("CpG_Site_distance","Average_of_adjusted_Pvalue")

data_plot %>% ggplot2::ggplot(aes(x= CpG_Site_distance, y=Average_of_adjusted_Pvalue)) +geom_line(color='steelblue', size=2 )+ ggtitle ("The distance of significant methylated CpG sites from TSS") + theme_minimal()

```

**The bar graph shows that the methylated CpG sites with significant P values are in the close distance from TSS. This result is consistent with previous data which showed that CpG sites are relatively enriched around transcription start sites of genes (Numata et al., 2012; Saxonov, Berg, & Brutlag, 2006).**

```{r}


x=tmp$mids;
y=histog;
data_plot=cbind(x,y)
data_plot=as.data.frame(data_plot)
colnames(data_plot)=c("CpG_site_distance","Average_of_adjusted_Pvalue")
data_plot %>% ggplot2::ggplot(aes(x= CpG_site_distance, y=Average_of_adjusted_Pvalue)) +geom_bar(stat="identity",fill="Green4")+ggtitle ("The distance of significant methylated CpG sites from TSS")+ theme_minimal()+theme_minimal() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))



```
### Conclusion:





























